gui/src/JobSelector.tsx

```diff
@@ -0,0 +1,711 @@
+    const displayEntityId = useDynamicProperty(props.displayEntityId, props.defaultDisplayEntityId, true);
+    const displaySubmitId = useDynamicProperty(props.displaySubmitId, props.defaultDisplaySubmitId, true);
+    const displayDate = useDynamicProperty(props.displayDate, props.defaultDisplayDate, true);
+
+    const headerToolbarSx = useMemo(
+        () => ({
+            pl: { sm: 2 },
+            pr: { xs: 1, sm: 1 },
+            ...(selected.length > 0 && {
+                bgcolor: "rgba(0, 0, 0, 0.05)",
+            }),
+        }),
+        [selected]
+    );
+
+    const jobSelectorColumns: JobSelectorColumns[] = useMemo(
+        () => [
+            {
+                id: "jobId",
+                primaryLabel: "Job",
+                showPrimaryLabel: displayJobId,
+                secondaryLabel: "ID",
+                showSecondayLabel: displayJobId,
+                columnIndex: 0,
+            },
+            {
+                id: "submitID",
+                primaryLabel: "Submit ID",
+                columnIndex: 4,
+                showPrimaryLabel: displaySubmitId,
+                showSecondayLabel: displaySubmitId,
+            },
+            {
+                id: "submitEntity",
+                primaryLabel: "Submitted entity",
+                secondaryLabel: "ID",
+                columnIndex: 3,
+                showPrimaryLabel: displayEntityLabel,
+                showSecondayLabel: displayEntityId,
+            },
+            {
+                id: "createdDt",
+                primaryLabel: "Creation date",
+                columnIndex: 5,
+                showPrimaryLabel: displayDate,
+                showSecondayLabel: displayDate,
+            },
+            {
+                id: "status",
+                primaryLabel: "Status",
+                columnIndex: 6,
+                showPrimaryLabel: true,
+                showSecondayLabel: true,
+            },
+        ],
+        [displayDate, displayEntityId, displayEntityLabel, displayJobId, displaySubmitId]
+    );
+
+    const handleClick = useCallback(
+        (event: React.MouseEvent<HTMLElement>, name: string) => {
+            const selectedIndex = selected.indexOf(name);
+            let newSelected: string[] = [];
+
+            if (selectedIndex === -1) {
+                newSelected = newSelected.concat(selected, name);
+            } else if (selectedIndex === 0) {
+                newSelected = newSelected.concat(selected.slice(1));
+            } else if (selectedIndex === selected.length - 1) {
+                newSelected = newSelected.concat(selected.slice(0, -1));
+            } else if (selectedIndex > 0) {
+                newSelected = newSelected.concat(selected.slice(0, selectedIndex), selected.slice(selectedIndex + 1));
+            }
+            setSelected([...newSelected]);
+        },
+        [selected]
```

Comment:
this doesn't need to depend on selected if we use the setState with function
ie: setSelected((oldSelected) => ...calculate new selected);